src/Compilers/CSharp/Portable/Symbols/Source/SourceUserDefinedOperatorSymbolBase.cs

```diff
@@ -97,6 +97,12 @@ protected SourceUserDefinedOperatorSymbolBase(
                 return;
             }
 
+            if (this.ContainingType is { IsExtension: true, ExtensionParameter.Type.IsStatic: true })
+            {
+                diagnostics.Add(ErrorCode.ERR_OperatorInExtensionOfStaticClass, location);
```

Comment:
nit: this feels somewhat redundant given existing `ERR_ParameterIsStaticClass`

Reply:
The same argument could be made for ```ERR_OperatorInStaticClass``` reported above. In addition, ```ERR_ParameterIsStaticClass``` is not reported for instance operators because there is no parameter to report.